src/index.rs

```diff
@@ -523,14 +545,18 @@ mod tests {
             kind: Kind::File,
             addrs: vec![],
             target: None,
+            umode: Default::default(),
+            owner: Default::default(),
         }];
         let index_json = serde_json::to_string(&entries).unwrap();
         println!("{}", index_json);
         assert_eq!(
             index_json,
             "[{\"apath\":\"/a/b\",\
              \"kind\":\"File\",\
-             \"mtime\":1461736377}]"
+             \"mtime\":1461736377,\
+             \"umode\":{\"mode\":33277},\
```

Comment:
It seems like we don't really need another layer of dict here, because the mode is really unlikely to have more than a u32... We could either:

1. Make `struct UnixMode(u32)`
2. Or tell serde to flatten it (https://serde.rs/attr-flatten.html)

I think I prefer 1...

Reply:
Cool, I didn't think of that. Changed UnixMode to a tuple struct: `UnixMode(u32)`